src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -100,5 +93,49 @@ static void CloseConnection(AmqpConnection connection)
 
             return connection;
         }
+
+        private TransportSettings CreateTransportSettings()
+        {
+            string hostName = this.Endpoint.Host;
+            string networkHost = this.Endpoint.Host;
+            int port = this.Endpoint.Port;
+            string scheme = this.Endpoint.Scheme;
+
+            if (scheme.Equals(WebSocketConstants.WebSocketSecureScheme, StringComparison.OrdinalIgnoreCase))
```

Comment:
>scheme.Equals(WebSocketConstants.WebSocketSecureScheme, StringComparison.OrdinalIgnoreCase) [](start = 16, length = 91)

Unfortunately, this is not how we decide whether to use web socket / tcp. 
Could you add `TransportType` as an enum in `ServiceBusConnectionStringBuilder` which has `Amqp` and `AmqpWebSockets` as the possible options. This has to be parsed from the connection string as a parameter and not by looking at the schema. Which also means, `CSB.ToString()` should appropriately output this parameter when TransportType is not set to default (AMQP).

There are couple of reasons for this call.
1. User might call it https instead of wss
2. (For a most common scenario) The SB namespace connection string from the portal contains sb:// as schema. So to use web sockets, he needs to manually append `TransportType=AmqpWebSockets` or change the schema. So both options require equal amount of work.
3. Changing an enum in `ConnectionStringBuilder` is easier and intuitive than changing the schema.
4. Maintaining compatibility with previous client.